### ISSUE TITLE
undefined method `signature'

### ISSUE THREAD

**Issue by vsuhachev** [dry-rb/dry-validation#215]
This code raises NoMethodError
``` ruby
require 'pp'
require 'dry-validation'

MySchema = Dry::Validation.Schema do
  required(:x).filled(:int?)

  rule(a: [:x]) do |x|
    x.gteq?(3)
  end

  rule(b: [:x]) do |x|
    x.gteq?(5)
  end
end

pp MySchema.call(x: 5).messages
pp MySchema.call(x: 4).messages
pp MySchema.call(x: 2).messages
```

**Comment by solnic** [comment]
Nice catch, thanks for reporting this. I'll try to fix it later this week and push 0.9.3.

**Comment by solnic** [comment]
Fixed in 0.9.3
